jte-models/src/main/java/gg/jte/models/runtime/JteModel.java

```diff
@@ -0,0 +1,30 @@
+package gg.jte.models.runtime;
+
+import gg.jte.Content;
+
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+public interface JteModel extends Content {
+    default void render(OutputStream outputStream, Charset charset) {
+        Writer writer = new OutputStreamWriter(outputStream, charset);
+        render(writer);
+    }
+
+    default void render(OutputStream outputStream) {
+        render(outputStream, StandardCharsets.UTF_8);
+    }
+
+    default String render() {
+        Writer writer = new StringWriter();
```

Comment:
`StringWriter` uses a synchronized `StringBuffer` under the hood and is rather slow.

I'm wondering if it wouldn't be better to have only one `render` method that accepts a `TemplateOutput` argument? This way users have full control over rendering if they need to. For instance, if they use precompiled UTF-8 templates, or have a special need for output escaping.

We could then still offer a convenience `render` method that provides a String by using `StringOutput`.

Reply:
I did not realise it would use a StringBuffer! I will definitely change that.

I think using TemplateOutput + convenience method will still meet my goals - I will try it.

Reply:
Done

Reply:
That looks really nice!